packages/js/src/plugins/auctionHouseModule/operations/buy.ts

```diff
@@ -0,0 +1,214 @@
+  useOperation,
+} from '@/types';
+import { SendAndConfirmTransactionResponse } from '../../rpcModule';
+import { AuctionHouse, Listing, Purchase } from '../models';
+
+// -----------------
+// Operation
+// -----------------
+
+const Key = 'BuyOperation' as const;
+
+/**
+ * Creates a bid on a given asset and then executes a sale on the created bid and listing.
+ *
+ * ```ts
+ * await metaplex
+ *   .auctionHouse()
+ *   .buy({ auctionHouse, listing, buyer })
+ *   .run();
+ * ```
+ *
+ * @group Operations
+ * @category Constructors
+ */
+export const buyOperation = useOperation<BuyOperation>(Key);
+
+/**
+ * @group Operations
+ * @category Types
+ */
+export type BuyOperation = Operation<typeof Key, BuyInput, BuyOutput>;
+
+/**
+ * @group Operations
+ * @category Inputs
+ */
+export type BuyInput = {
+  /** The Auction House in which to create a Bid and execute a Sale. */
+  auctionHouse: AuctionHouse;
+
+  /**
+   * The Auction House authority.
+   * If this is Signer the transaction fee
+   * will be paid from the Auction House Fee Account
+   *
+   * @defaultValue `auctionHouse.authority`
+   */
+  authority?: PublicKey | Signer;
+
+  /**
+   * Creator of a bid.
+   *
+   * @defaultValue `metaplex.identity()`
+   */
+  buyer?: Signer;
+
+  /**
+   * The Listing that is used in the sale.
+   * We only need a subset of the `Listing` model but we
+   * need enough information regarding its settings to know how
+   * to execute the sale.
+   *
+   * This includes, its asset, auction house address, seller, receipt address etc.
+   */
+  listing: Pick<
+    Listing,
+    | 'asset'
+    | 'auctionHouse'
+    | 'canceledAt'
+    | 'sellerAddress'
+    | 'tradeStateAddress'
+    | 'receiptAddress'
+  >;
+
+  /**
+   * The Auctioneer authority key.
+   * It is required when Auction House has Auctioneer enabled.
+   *
+   * @defaultValue No default value.
+   */
+  auctioneerAuthority?: Signer;
+
+  /**
+   * The token account address that's associated to the asset a bid created is for.
+   * If this or seller isn't provided in the Listing, then the bid will be public.
+   *
+   * @defaultValue No default value.
+   */
+  tokenAccount?: Option<PublicKey>;
+
+  /**
+   * The buyer's price.
+   *
+   * @defaultValue 0 SOLs or tokens.
+   */
+  price?: SolAmount | SplTokenAmount;
+
+  /**
+   * The number of tokens to bid for.
+   * For an NFT bid it must be 1 token.
+   *
+   * When a Fungible Asset is put on sale.
+   * The buyer can then create a buy order of said assets that is
+   * less than the token_size of the sell order.
+   *
+   * @defaultValue 1 token.
+   */
+  tokens?: SplTokenAmount;
+
+  /**
+   * Prints the bid receipt.
+   * The receipt holds information about the bid,
+   * So it's important to print it if you want to use the `Bid` model
+   *
+   * The receipt printing is skipped for the Auctioneer Auction House
+   * Since it currently doesn't support it.
+   *
+   * @defaultValue `true`
+   */
+  printReceipt?: boolean;
```

Comment:
The comment should explain that this will be used for both printing the bid and the purchase receipts.